service/controller/v3/resource/vpngateway/delete.go

```diff
@@ -61,27 +62,34 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, azureConfig, change in
 	return nil
 }
 
-func (r *Resource) applyDeleteChange(ctx context.Context, azureConfig providerv1alpha1.AzureConfig, change network.VirtualNetworkPeering) error {
-	r.logger.LogCtx(ctx, "level", "debug", "message", "deleting host vnet peering")
+func (r *Resource) applyDeleteChange(ctx context.Context, azureConfig providerv1alpha1.AzureConfig, change connections) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring host vpn gateway connection is deleted")
 
-	vnetPeeringClient, err := r.getVnetPeeringClient()
+	if change.isEmpty() {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensured host vpn gateway connection is deleted")
+		return nil
+	}
+
+	hostGatewayConnectionClient, err := r.getHostVirtualNetworkGatewayConnectionsClient(ctx)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	respFuture, err := vnetPeeringClient.Delete(ctx, r.azure.HostCluster.ResourceGroup, r.azure.HostCluster.ResourceGroup, *change.Name)
+	resourceGroup := r.azure.HostCluster.ResourceGroup
+	connectionName := *change.Host.Name
+
+	respFuture, err := hostGatewayConnectionClient.Delete(ctx, resourceGroup, connectionName)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	res, err := vnetPeeringClient.DeleteResponder(respFuture.Response())
+	res, err := hostGatewayConnectionClient.DeleteResponder(respFuture.Response())
 	if client.ResponseWasNotFound(res) {
```

Comment:
Why is it actually checked? I mean if there is a response when we don't care if there isn't.

Reply:
done